File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>ans;
        unordered_map<int,int>freq;
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            int req=target-nums[i];
            if(freq.find(req) != freq.end())
            {
                ans.push_back(i);
                ans.push_back(freq[req]);
            }
            else
                freq[nums[i]]=i;
        }
        return ans;
        
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const n=nums.length;
    let mp=new Map();
    for(let i=0;i<n;i++)
    {
        let req=target-nums[i];
       if(mp.has(req))
        {
            return [i,mp.get(req)];
        }
        else
        {
            mp.set(nums[i],i);
        }
    }
    return -1;

    
};